third_party/opentelemetry-proto/Protobuf.cmake

```diff
@@ -1,48 +1,68 @@
+include(FetchContent)
+FetchContent_Declare(
+  gRPC
+  GIT_REPOSITORY https://github.com/grpc/grpc
+  GIT_TAG v1.32.0)
+FetchContent_MakeAvailable(gRPC)
```

Comment:
As far as I know this requires CMake 3.14+, we currently have a minimum requirement of 3.1.

Could you please increase the required CMake version in the global `CMakeLists.txt` file? Then we can review this together with this PR.

Reply:
Could upgrade CMake to 3.14 causes compatibility issue? In my Ubuntu 18.04 LTS, the default installed one is 3.10.

Reply:
It'll take 15 minutes of CI build time to build latest CMake from source for Ubuntu 14.xx and Ubuntu 18.xx

Reply:
I added this script here: https://github.com/open-telemetry/opentelemetry-cpp/blob/master/tools/setup-cmake.sh and the best scenario to run it while creating docker image, to subsequently reuse that docker image (that's how I build locally). We need to be creative for CI. Either publishing docker image or using GitHub cache action (5GB cache) to pre-cache an image that is built once, then we save 15 mins of CMake buld time.... I assume we do not want to use some Ubuntu PPA to host the latest CMake compiled for old Ubuntu..